### PR TITLE
make QPS and Burst values configurable via pilot features

### PR DESCRIPTION
**Pull request by nirvanagit**
Make QPS and Burst values configurable via pilot features

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[X] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Reference - 
- [1.5 port](https://github.com/istio/istio/pull/25660)
- [1.6 port](https://github.com/istio/istio/pull/25661)

### CODE REVIEW

File: pilot/pkg/bootstrap/options.go [hunk 1]
```diff
@@ -151,4 +151,6 @@ func (p *PilotArgs) applyDefaults() {
 	p.KeepaliveOptions = keepalive.DefaultOption()
 	p.RegistryOptions.DistributionTrackingEnabled = features.EnableDistributionTracking
 	p.RegistryOptions.DistributionCacheRetention = features.DistributionHistoryRetention
+	p.RegistryOptions.KubeOptions.KubernetesAPIQPS = float32(features.KubernetesAPIQPS.Get())
```
Comment (rshriram) [inline pilot/pkg/bootstrap/options.go:154]: make this as cli args please rather than features

Comment (nirvanagit) [reply pilot/pkg/bootstrap/options.go]: Done

File: pilot/cmd/pilot-discovery/main.go [hunk 2]
```diff
@@ -177,6 +177,12 @@ func init() {
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.ServerOptions.TLSOptions.KeyFile, "tlsKeyFile", "",
 		"File containing the x509 private key matching --tlsCertFile")
 
+	discoveryCmd.PersistentFlags().Float32Var(&serverArgs.RegistryOptions.KubeOptions.KubernetesAPIQPS, "kubernetesApiQPS", 20.0,
+		"Maximum QPS when communicating with the kubernetes API")
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-discovery/main.go:181]: Add default value to the msg

**Pull request merged**